src/pyopticon/dashboard.py

```diff
@@ -152,6 +157,39 @@ def add_interlock(self, fn):
         :type fn: function
         """
         self.all_interlocks.append(fn)
+
+    def notify(self, event, modifier =None):
+        """ Updates observers when there is an event logged. If an observer created the event, they are not updated.
+
+        :param modifier: the observer that created an event
+        :type modifier: any widget
+        """
+        print(event)
+        for observer in self._observers:
+            if modifier != observer:
+                print("Notified: " + str(observer))
+                observer.update(event)
+
+    def register_observer(self, observer):
+        """If an observer is not on the list, append it to the list.
+
+        :param observer: a widget that needs to receive updates from the publisher
+        :type observer: any widget
+        """
+        if observer not in self._observers:
+            self._observers.append(observer)
+
+    def detach_observer(self, observer):
```

Comment:
Similar to above, `remove_observer()` might be a bit more clear? `detach` feels a bit unnecessarily technical.